csp_billing_adapter/bill_utils.py

```diff
@@ -201,6 +204,65 @@ def get_billing_dimensions(
     return billed_dimensions
 
 
+def filter_usage_records_by_date_range(
+    usage_records: list,
+    range_start: str,
+    range_end: str
+) -> list:
+    """
+    Returns the list of 'usage_records' filtered to remove records
+    that fall outside the specified, inclusive, date range, from
+    'range_start' to 'range_end'.
+
+    :param usage_records: The list of usage records to be filtered.
+    :param range_start: The start of the inclusive date range.
+    :param range_end: The end of the inclusive date range.
+    :return:
+        The set of usage records that fails within the specified
+        date range.
+    """
+    return [record for record in usage_records
+            if range_start <= record['reporting_time'] <= range_end]
```

Comment:
I think one condition can be exclusive. Otherwise there's technically overlap if a timestamp happens to match exactly. Chances are slim but it's easy enough to prevent. I'd suggest the range end is < as opposed to <=.